lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -2835,6 +2835,94 @@ OpFoldResult AtenDivIntOp::fold(FoldAdaptor adaptor) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// AtenIndexSelectOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenIndexSelectOp::fold(FoldAdaptor adaptor) {
+  auto self = getSelf();
+  auto index = getIndex();
+  auto selfTy = cast<ValueTensorType>(self.getType());
```

Comment:
I wasn't sure about asserting that the type is ValueTensorType, I asked a q here: https://discord.com/channels/636084430946959380/742573221882364009/1204378906925342750

Reply:
You are right!  As it turns out, during a torch.tensor's journey to a shaped, typed torch.vtensor, the folders (and I assume everything else in canonicalization) gets called repeatedly on each step, so it appears that the correct strategy is to keep rejecting any attempts to fold until the tensors' shapes and types are nice.  The new code reflects that behavior.